pkg/iac/terraform/state/terraform_state_reader.go

```diff
@@ -165,16 +172,20 @@ func (r *TerraformStateReader) convertInstance(instance *states.ResourceInstance
 	return instanceObj, nil
 }
 
-func (r *TerraformStateReader) decode(values map[string][]cty.Value) ([]resource.Resource, error) {
+func (r *TerraformStateReader) decode(valFromState map[string][]decodedRes) ([]resource.Resource, error) {
 	results := make([]resource.Resource, 0)
 
-	for ty, val := range values {
-		decodedResources, err := r.deserializer.Deserialize(ty, val)
-		if err != nil {
-			logrus.WithField("ty", ty).Warnf("Could not read from state: %+v", err)
-			continue
+	for ty, val := range valFromState {
+		for _, stateVal := range val {
+			res, err := r.deserializer.DeserializeOne(ty, stateVal.val)
+			if err != nil {
+				logrus.WithField("ty", ty).Warnf("Could not read from state: %+v", err)
```

Comment:
Maybe we can display here in which state the error came from

Reply:
Yeah good idea, that was not the original goal of this PR but it can't hurt. Let's do this !